mlrun/api/api/endpoints/tags.py

```diff
@@ -0,0 +1,147 @@
+import sqlalchemy.orm
+
+import mlrun.api.api.deps
+import mlrun.api.crud.tags
+import mlrun.api.schemas
+import mlrun.api.utils.auth.verifier
+import mlrun.api.utils.singletons.project_member
+
+router = fastapi.APIRouter()
+
+
+@router.post("/projects/{project}/tags/{tag}")
+async def overwrite_object_tags_with_tag(
+    project: str,
+    tag: str,
+    objects: mlrun.api.schemas.TagsObjects,
+    auth_info: mlrun.api.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: sqlalchemy.orm.Session = fastapi.Depends(
+        mlrun.api.api.deps.get_db_session
+    ),
+):
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().ensure_project,
+        db_session,
+        project,
+        auth_info=auth_info,
+    )
+
+    for object_list in objects.objects:
+        # check permission per object type
+        await fastapi.concurrency.run_in_threadpool(
+            mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions,
+            getattr(mlrun.api.schemas.AuthorizationResourceTypes, object_list.kind),
+            project,
+            resource_name=None,
+            # not actually overwriting objects, just overwriting the objects tags
+            action=mlrun.api.schemas.AuthorizationAction.update,
+            auth_info=auth_info,
+        )
+
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.crud.Tags().overwrite_object_tags_with_tag,
+        db_session,
+        project,
+        tag,
+        objects.objects,
+    )
+    return {}
+
+
+@router.put("/projects/{project}/tags/{tag}")
+async def append_tag_to_objects(
+    project: str,
+    tag: str,
+    objects: mlrun.api.schemas.TagsObjects,
+    auth_info: mlrun.api.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: sqlalchemy.orm.Session = fastapi.Depends(
+        mlrun.api.api.deps.get_db_session
+    ),
+):
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().ensure_project,
+        db_session,
+        project,
+        auth_info=auth_info,
+    )
+
+    for object_list in objects.objects:
+        # check permission per object type
+        await fastapi.concurrency.run_in_threadpool(
+            mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions,
+            getattr(mlrun.api.schemas.AuthorizationResourceTypes, object_list.kind),
+            project,
+            resource_name=None,
+            action=mlrun.api.schemas.AuthorizationAction.update,
+            auth_info=auth_info,
+        )
+
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.crud.Tags().append_tag_to_objects,
+        db_session,
+        project,
+        tag,
+        objects.objects,
+    )
+    return {}
+
+
+@router.delete("/projects/{project}/tags/{tag}")
```

Comment:
```suggestion
@router.delete("/projects/{project}/tags/{tag}", status_code=HTTPStatus.NO_CONTENT.value)
```

Since you're not returning anything, in `DELETE` we usually return 204. Adding this property in the decorator means you don't have to return anything, and this status will be automatically returned (unless you return something else or assert which will translate to whatever REST value the assertion maps to)